kornia/filters/sobel.py

```diff
@@ -52,7 +52,10 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
 
         # convolve input tensor with sobel kernel
         kernel_flip: torch.Tensor = kernel.flip(-3)
-        return F.conv3d(input[:, :, None], kernel_flip, padding=1, groups=c)
+        # Pad with "replicate for spatial dims, but with zeros for channel dim
+        padded_inp = F.pad(F.pad(input, [1, 1, 1, 1], 'replicate')[:, :, None],
+                           [0, 0, 0, 0, 1, 1], 'constant')
```

Comment:
I would explicitly add the constant value to make it more readable. I guess is zero